conan/tools/microsoft/visual.py

```diff
@@ -132,12 +132,14 @@ def generate(self, scope="build"):
             vcvars_ver = _vcvars_vers(conanfile, compiler, vs_version)
         vcvarsarch = _vcvars_arch(conanfile)
 
+        winsdk_version = conanfile.conf.get("tools.microsoft:winsdk_version", check_type=str)
+        winsdk_version = winsdk_version or conanfile.setttings.get_safe("os.version")
```

Comment:
I think this might cause issues if an SDK that matches the os version is not installed - but also, I would advise against making this the default.
Last time I checked, the default behaviour of vcvars was to pick up "the most recent" version, unless told otherwise - I'd try and keep that behaviour. We did have a a lot of issues in Conan Center when CMake was picking up a version that matches the os.version and not the latest. Recently CMake has added a policy to NOT behave like that: https://cmake.org/cmake/help/latest/policy/CMP0149.html#policy:CMP0149

Reply:
This will only be used if:
- Users adds ``version`` to Windows subsettings in ``settings.yml``
- Defines ``os.version`` in their profiles

I have kept ``os.version`` instead of ``sdk_version`` because other Windows oss, like CE in ``settings.yml`` already have a ``version`` subsetting that represents the sdk

Reply:
Added integrations for VCVars, MSBuildToolchain (so opening directly from IDE works) and CMakeToolchain